test/test_account.py

```diff
@@ -0,0 +1,188 @@
+"""
+Test account functionality.
+"""
+
+import pytest
+
+from starkware.cairo.common.hash_state import compute_hash_on_elements
+from starkware.crypto.signature.signature import private_to_stark_key, sign
+from starkware.starknet.public.abi import get_selector_from_name
+
+from .shared import ABI_PATH, CONTRACT_PATH
+from .util import assert_tx_status, deploy, run_devnet_in_background, call, invoke
+
+ACCOUNT_ARTIFACTS_PATH = "starknet_devnet/accounts_artifacts"
+ACCOUNT_AUTHOR = "OpenZeppelin"
+ACCOUNT_VERSION = "0.1.0"
+
+ACCOUNT_PATH = f"{ACCOUNT_ARTIFACTS_PATH}/{ACCOUNT_AUTHOR}/{ACCOUNT_VERSION}/Account.cairo/Account.json"
+ACCOUNT_ABI_PATH = f"{ACCOUNT_ARTIFACTS_PATH}/{ACCOUNT_AUTHOR}/{ACCOUNT_VERSION}/Account.cairo/Account_abi.json"
+SALT = "0x99"
+ACCOUNT_ADDRESS = "0x066a91d591d5ba09d37f21fd526242c1ddc6dc6b0ce72b2482a4c6c033114e3a"
+
+TRANSACTION_VERSION = 0
+
+PRIVATE_KEY = 123456789987654321
+PUBLIC_KEY = private_to_stark_key(PRIVATE_KEY)
+
+@pytest.fixture(autouse=True)
+def run_before_and_after_test():
+    """Cleanup after tests finish."""
+
+    # before test
+    devnet_proc = run_devnet_in_background()
+
+    yield
+
+    # after test
+    devnet_proc.kill()
+
+def deploy_empty_contract():
+    """Deploy sample contract with balance = 0."""
+    return deploy(CONTRACT_PATH, inputs=["0"], salt=SALT)
+
+def deploy_account_contract():
+    """Deploy account contract."""
+    return deploy(ACCOUNT_PATH, inputs=[str(PUBLIC_KEY)], salt=SALT)
+
+def get_nonce():
+    """Get nonce."""
+    return call("get_nonce", ACCOUNT_ADDRESS, ACCOUNT_ABI_PATH)
+
+def get_execute_calldata(call_array, calldata, nonce):
+    """Get calldata for __execute__."""
+    return [
+        len(call_array),
+        *[x for t in call_array for x in t],
+        len(calldata),
+        *calldata,
+        int(nonce)
+    ]
+
+def str_to_felt(text):
+    """Converts string to felt."""
+    return int.from_bytes(bytes(text, "ascii"), "big")
+
+def hash_multicall(sender, calls, nonce, max_fee):
+    """desc"""
+    hash_array = []
+
+    for call_tuple in calls:
+        call_elements = [call_tuple[0], call_tuple[1], compute_hash_on_elements(call_tuple[2])]
+        hash_array.append(compute_hash_on_elements(call_elements))
+
+    return compute_hash_on_elements([
+        str_to_felt('StarkNet Transaction'),
+        sender,
+        compute_hash_on_elements(hash_array),
+        nonce,
+        max_fee,
+        TRANSACTION_VERSION
+    ])
+
+
+def get_signature(message_hash):
+    """Get signature from message hash and private key."""
+    print(f"message hash {message_hash}")
+
+    sig_r, sig_s = sign(message_hash, PRIVATE_KEY)
+    return [str(sig_r), str(sig_s)]
+
+def from_call_to_call_array(calls):
+    """Transforms calls to call_array and calldata."""
+    call_array = []
+    calldata = []
+
+    for call_tuple in calls:
+        assert len(call_tuple) == 3, "Invalid call parameters"
+
+        entry = (call_tuple[0], get_selector_from_name(
```

Comment:
Can this look better?

Reply:
[6d2a3f4](https://github.com/Shard-Labs/starknet-devnet/pull/69/commits/6d2a3f49b02db978fd0034b814f35f7ed3ae651d)